bootstrap/bootstrap.go

```diff
@@ -108,6 +115,10 @@ func Run(manifest []byte, ch chan<- *StepInfo, minHosts int) (err error) {
 	state := &State{
 		StepData:  make(map[string]interface{}),
 		Providers: make(map[string]*ct.Provider),
+		Singleton: minHosts == 1,
+	}
+	if s := os.Getenv("SINGLETON"); s != "" {
+		state.Singleton = s == "true"
```

Comment:
I'm not sure that would work, as it would override the `minHosts == 1` check above even if unset.